src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -180,6 +181,35 @@ public interface OnHitEffect
 		}
 	};
 	
+	/**
+	 * A knockback effect that does a right-vector (relative to the player's orientation) and direction to enemy dot calculation (relative to the player's orientation) to spread enemies horizontally via knockback.
+	 */
+	OnHitEffect SPREADING_KNOCKBACK = (stack, target, attacker) -> {
+		
+		//Attacker's right direction and position
+		double pRightSin = Math.sin(attacker.getYRot() * ((float) Math.PI / 180F) - Math.PI / 2.0);
+		double pRightCos = Math.cos(attacker.getYRot() * ((float) Math.PI / 180F) - Math.PI / 2.0);
```

Comment:
A vector could ideally be used for these as well.

Reply:
Change made, replaced instances of Vec2 with Vec3 for some more readability (use z component instead of y) and more built-in functions, which also reduced the amount of local variables.